test/pg_handler/bench_test.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"os"
+	"fmt"
+	"testing"
+	"database/sql"
+	_ "github.com/lib/pq"
+)
+
+const (
+	socketPath = "/sock/.s.PGSQL.5432"
+	select1Query = "SELECT 1"
+	select10Query = "SELECT generate_series(0, 9)"
+	select100Query = "SELECT generate_series(0, 99)"
+	select1000Query = "SELECT generate_series(0, 999)"
+	select10000Query = "SELECT generate_series(0, 9999)"
+	envPostgresPort = "POSTGRES_PORT"
+	envSecretlessPort = "SECRETLESS_PORT"
+	dockerPortPostgres = "5432"
+	dockerPortSecretless = "15432"
+)
+
+func getPostgresConnection() (*sql.DB, error) {
+	port := os.Getenv(envPostgresPort)
+	if port == "" {
+		port = dockerPortPostgres
+	}
+	connStr := fmt.Sprintf("port=%s sslmode=disable database=postgres user=test", port)
+	return configureDB(sql.Open("postgres", connStr))
+}
+
+func getSecretlessConnection() (*sql.DB, error) {
+	port := os.Getenv(envSecretlessPort)
+	if port == "" {
+		port = dockerPortSecretless
+	}
+	connStr := fmt.Sprintf("port=%s sslmode=disable database=postgres", port)
+	return configureDB(sql.Open("postgres", connStr))
+}
```

Comment:
Not a blocker but the connection setup of these two seems like a good dedupe candidate